src/services/users.service.ts

```diff
@@ -52,6 +52,17 @@ class UserService {
     const deleteUserData = await this.users.delete({ where: { id: userId }, select: { email: true, id: true } });
     return deleteUserData;
   }
+
+  public async manageGroups(userId: number, data: any) {
+    if (isEmpty(data)) throw new HttpException(400, "Groups don't exist");
```

Comment:
Ensure that the data parameter passed to the manageGroups method is properly validated to prevent unexpected behavior. You can use validation libraries like class-validator or manually validate the input data before performing the update operation.